src/hdf5_hl.ts

```diff
@@ -677,14 +665,27 @@ abstract class HasAttrs {
     return new Reference(ref_data);
   }
 
-  dereference(ref: Reference | RegionReference) {
+  dereference(ref: Reference | RegionReference): DatasetRegion | getReturnTypes | null {
     const is_region = (ref instanceof RegionReference);
```

Comment:
(I can make a note to do this in a later PR.)

Reply:
`RegionReference` doesn't extend `HasAttrs`, so what I suggested was not possible. Instead, I added signature overrides to the `dereference` method so that if one passes a `RegionReference`, the return type is `DatasetRegion`:

```
dereference(ref: RegionReference): DatasetRegion;
dereference(ref: Reference | RegionReference): DatasetRegion | getReturnTypes | null;
```